### PR TITLE
[New Template] vpc/vpc-flow-logs-s3 - Deliver VPC Flow Logs to S3

### PR DESCRIPTION
**Pull request by michaelwittig**
[Improvement] state/s3 - Additional access capability: FlowLogWrite

### CODE REVIEW

File: vpc/vpc-flow-logs-s3.yaml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        default: 'Flow Logs Parameters'
+      Parameters:
+      - ExternalLogBucket
+      - LogFilePrefix
+      - TrafficType
+Parameters:
+  ParentVPCStack:
+    Description: 'Stack name of parent VPC stack based on vpc/vpc-*azs.yaml template.'
+    Type: String
+  ExternalLogBucket:
```
Comment (andreaswittig) [inline vpc/vpc-flow-logs-s3.yaml]: Why not referencing a ` state/s3` stack? For increased flexibility?

File: state/s3.yaml [hunk 2]
```diff
@@ -227,6 +322,52 @@ Resources:
...
+          Condition:
+            StringEquals:
+              's3:x-amz-acl': 'bucket-owner-full-control'
+        - Sid: AWSLogDeliveryAclCheck
+          Effect: Allow
+          Principal:
+            Service: 'delivery.logs.amazonaws.com'
+          Action: 's3:GetBucketAcl'
+          Resource: !GetAtt 'Bucket.Arn'
+  BucketPolicyFlowLogCrossAccountWrite:
```
Comment (andreaswittig) [inline state/s3.yaml]: This allows any AWS account to publish their flow logs to your AWS account, correct? Is there no way to restrict based on OU/Account Id/...?

File: state/s3.yaml [hunk 3]
```diff
@@ -213,12 +286,34 @@ Resources:
           Condition:
             StringEquals:
               's3:x-amz-acl': 'bucket-owner-full-control'
+  BucketPolicyCloudTrailCrossAccountWrite:
```
Comment (andreaswittig) [inline state/s3.yaml]: This allows any AWS account to publish their logs to your AWS account, correct? Is there no way to restrict based on OU/Account Id/...?

File: state/s3.yaml [hunk 4]
```diff
@@ -189,7 +240,28 @@ Resources:
           Principal:
             Service: 'config.amazonaws.com'
           Action: 's3:PutObject'
-          Resource: !Sub '${Bucket.Arn}/*AWSLogs/${AWS::AccountId}/Config/*'
+          Resource: !Sub '${Bucket.Arn}/*AWSLogs/${AWS::AccountId}/*'
+          Condition:
+            StringEquals:
+              's3:x-amz-acl': 'bucket-owner-full-control'
+  BucketPolicyConfigCrossAccountWrite:
```
Comment (andreaswittig) [inline state/s3.yaml]: This allows any AWS account to publish their logs to your AWS account, correct? Is there no way to restrict based on OU/Account Id/...?

File: vpc/vpc-flow-logs-s3.yaml (hunk 1 above)
Comment (michaelwittig) [reply vpc/vpc-flow-logs-s3.yaml]: yes. for multi account setups.

File: state/s3.yaml (hunk 2 above)
Comment (michaelwittig) [reply state/s3.yaml]: I made an experiment with the condition `aws:PrincipalOrgID` but it doesn't seem to work. Let me open a support case and see what AWS suggests...

**Comment by michaelwittig** [comment]
wait for AWS Support

File: state/s3.yaml (hunk 2 above)
Comment (michaelwittig) [reply state/s3.yaml]: removed

File: state/s3.yaml (hunk 3 above)
Comment (michaelwittig) [reply state/s3.yaml]: removed

**Pull request merged**